File: Csharp2/MyLexer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using bill.minipython5_indented.lexer;
using bill.minipython5_indented.node;
using bill.minipython5_indented.parser;
using MiniPython1.walkers;
using System.Text.RegularExpressions;

namespace MiniPython1
{
    class MyLexer : Lexer
    {
        private int count;
        private TComment comment;
        private StringBuilder text;


        public MyLexer(TextReader tr)
            : base(tr)
        {

        }


        protected override void Filter()
        {
            if (this.currentState.Equals(State.COMMENT))
            {
                if (comment == null)
                {
                    comment = (TComment)token;
                    text = new StringBuilder(comment.Text);
                    count = 1;
                    token = null;

                }
                else
                {
                    text.Append(token.Text);
                    if (token is TComment)
                    {
                        count++;
                    }
                    else if (token is TNewLine)
                    {
                        count--;
                    }
                    if (count != 0)
                    {
                        token = null;
                    }
                    else
                    {
                        comment.Text = text.ToString();
                        token = comment;
                        this.currentState = State.NORMAL;
                        comment = null;
                    }
                }
            }

        }
    }
}

File: Java_parser/MyLexer.java
import bill.minipython5_indented.lexer.Lexer;
import bill.minipython5_indented.node.Start;
import bill.minipython5_indented.node.Token;
import bill.minipython5_indented.parser.Parser;
import bill.minipython5_indented.*;
import bill.minipython5_indented.lexer.Lexer.State;
import bill.minipython5_indented.node.*;

public class MyLexer extends Lexer {
	private int count;
	private TComment comment;
	private StringBuffer text;

	// We define a constructor
	public MyLexer(java.io.PushbackReader in) {
		super(in);
	}

	// We define a filter that recognizes nested comments.
	protected void filter() { // if we are in the comment state
		if (state.equals(State.COMMENT)) { // if we are just entering this state
			if (comment == null) { // The token is supposed to be a comment.
									// We keep a reference to it and set the
									// count to one
				comment = (TComment) token;
				text = new StringBuffer(comment.getText());
				count = 1;
				token = null; // continue to scan the input.
			} else { // we were already in the comment state
				text.append(token.getText()); // accumulate the text.
				if (token instanceof TComment)
					count++;
				else if (token instanceof bill.minipython5_indented.node.TNewLine)
					count--;
				if (count != 0)
					token = null; // continue to scan the input.
				else {
					comment.setText(text.toString());
					token = comment; // return a comment with the full text.
					state = State.NORMAL; // go back to normal.
					comment = null; // We release this reference.
				}
			}
		}
	}

}
